metrics/clients_test.go

```diff
@@ -53,49 +73,49 @@ func StartMockGrpcService() {
 	_ = <-ch
 }
 
-func Test_callgRPCServiceHeartbeat(t *testing.T) {
+func (suite *ClientTestSuite) Test_callgRPCServiceHeartbeat() {
 	// Start the grpc mock server
 	StartMockGrpcService()
 
 	serviceURL := "localhost" + testPort
 	heartbeat, err := callgRPCServiceHeartbeat(serviceURL)
-	assert.False(t, err != nil)
+	assert.False(suite.T(), err != nil)
 
-	assert.NotEqual(t, `{}`, string(heartbeat), "Service Heartbeat must not be empty.")
-	assert.Equal(t, heartbeat.String(), pb.HealthCheckResponse_SERVING.String())
+	assert.NotEqual(suite.T(), `{}`, string(heartbeat), "Service Heartbeat must not be empty.")
+	assert.Equal(suite.T(), heartbeat.String(), pb.HealthCheckResponse_SERVING.String())
 
 	serviceURL = "localhost:26000"
 	heartbeat, err = callgRPCServiceHeartbeat(serviceURL)
-	assert.True(t, err != nil)
+	assert.True(suite.T(), err != nil)
 }
 
-func Test_callHTTPServiceHeartbeat(t *testing.T) {
-	serviceURL := "http://demo5343751.mockable.io/heartbeat"
+func (suite *ClientTestSuite) Test_callHTTPServiceHeartbeat() {
+	serviceURL := "http://localhost:1111/heartbeat"
```

Comment:
Lets make the port dynamic - via config file etc

Reply:
have removed the hardcoded endpoint from multiple places, the service url end point is specified as a config file , for a test case , dont wanna add a new config for the port number

Reply:
[e949ae6](https://github.com/singnet/snet-daemon/pull/566/commits/e949ae6e5b0789b26553a46c7d341f25a6cb082e) - latest commit